src/robotFactory.js

```diff
@@ -1,10 +1,99 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+    this.chipVersion = chipVersion;
+  };
 
-class DeliveryDrone {}
+  getInfo() {
+    return `Robot: ${this.name}, `
+    + `Chip version: ${this.chipVersion}, `
+    + `Weight: ${this.weight}`;
+  };
+
+  goForward(step = 1) {
+    this.coords.y += step;
+
+    return this;
+  };
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+
+    return this;
+  };
+
+  goRight(step = 1) {
+    this.coords.x += step;
+
+    return this;
+  };
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+
+    return this;
+  };
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+      z: coords.z || 0,
+    };
+  };
+
+  goUp(step = 1) {
+    this.coords.z += step;
+
+    return this;
+  };
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+
+    return this;
+  };
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name,
+    weight,
+    coords,
+    chipVersion,
+    maxLoadWeight,
+    currentLoad = null) {
```

Comment:
if you multi-line, stick to the rule of aligned open and closing parenthesis
```suggestion
  constructor(
    name,
    weight,
    coords,
    chipVersion,
    maxLoadWeight,
    currentLoad = null,
  ) {
```